working-groups/meta/rfc-drafts/compiler-team-contributors.md

```diff
@@ -0,0 +1,244 @@
+months, then we will ask them if they would like to go to alumni
+status. If they respond yes or do not respond, they can be placed on
+alumni status.  If they would prefer to remain active, that is also
+fine, but they will get asked again periodically if they continue to
+be inactive.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+This RFC represents, effectively, the smallest extension to our structure
+that could possibly work. One could imagine more elaborate structures along a few dimensions.
+
+**More senior levels of membership.** One concern is that the set of
+**members** of the compiler team may grow too large for things like
+FCP (where each person must check their box) to be feasible. This
+could be resolved by moving away from FCP-based decision making (which
+is rarely used in the compiler anyhow), but it may also be worth
+considering another level of membership (e.g., a **senior
+member"). Senior members could be used for FCP-level decisions, which
+would presumably be relatively rare. At present there is a kind of
+implicit amount of "seniority" amongst members, where the opinions of
+people who have been around for longer are obviously given great
+weight, but formalizing this could have value.
+
+**Specialists and organizers.** Right now, we don't draw a distinction
+between people who write code and those who (for example) perform more
+organizational roles (as of the time of this writing, we don't have
+any members who perform more organizational roles exclusively, but
+that is a likely future development). There will definitely be
+contributors who would rather not participate in the more
+organizational aspects of running the team, but would prefer to simply
+write code. As the team gets more and more organized, it seems likely
+that we may want to recognize this distinction, just to avoid things
+like pinging folks with organizational questions when they are not
+interested in that. But we could also address this by growing more
+kinds of groups within the set of members, such that one rarely pings
+the full set of members.
+
+# Prior art
+[prior-art]: #prior-art
+
+It would be good to include some survey of how other open source
+organizations manage themselves here.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+**Are "contributor" and "member" the best names to use?** The term
+"member" is used pretty universally amongst subteams to refer to
+"decision makers", so I wanted to stick to it, but I was tempted by
+other terms like "member" and "senior member".
+
+**What set of privileges should be retained in alumni status?** For
+example, should you still have r+ privileges? I'm inclined to say no.
+
+**What level of inactivity merits one for alumni status?** The RFC
+presently says 6 months, but that number was pulled out of a
+(metaphorical) hat.
```

Comment:
I guess all that depends, did it happen already that someone was in alumni and came back to member status again?. What happens if somebody stays on the team being inactive for more than 6 months? does that cause some kind of trouble or something?. My guess is that 6 months seems like not a lot of time in case someone decides to take a break but if you can come back easily I also think it's an ok period of time too :).

Reply:
> What happens if somebody stays on the team being inactive for more than 6 months?

I think they would just continue being alumni. I think this is just a way to distinguish between active and inactive members, but also to still recognize people who were actively participating as members and for whatever reasons haven't continued to do so, and to give these people an easy way to get back involved.

Reply:
So far, nobody has ever come back from alumni status -- but we definitely have the problem of people getting r+ status and then silently disappearing for one reason or another. My intention with these clauses is not to force anyone out, but just to have some mechanism to try and keep the list of contributors etc reasonably accurate. 

I definitely want the ability for people to come back from alumni status, though -- I don't think that becoming active or inactive needs to be a "weighty" decision.

Reply:
I'm interested in the situation "previously contributed often, now contributes rarely and isn't "actively helping to shape the direction of the team", but still contributes more often than once in 6 months".

I kind of thought about going alumni, but I'm still doing reviews and plan to make PRs occasionally, so I wouldn't want to lose r+ rights.

Reply:
I think this is a really good point. We should try to find ways that alumni or team members who don’t have a lot of time but still want to be involved are able to do things like reviews and share the knowledge and experience they’ve accumulated with the compiler.

Reply:
So, I added this text in the Future Possibilities section: 

> We may want to refine the notion of alumni. In particular, members may
> wish to drop back to contributor level without becoming full alumni,
> and this RFC doesn't account for that (but it also doesn't preclude
> it).

but really, maybe I should just add that as an explicit possibility? I don't see any reason it's a problem. And presumably said people could come back to full status at any point (just like any alumni). Would that seem good to you, @petrochenkov ?

Reply:
I went ahead and added that text. I also added a caveat that rejoining from alumni status etc is "normally automatic barring extraordinary circumstances". I can't really imagine there every being a problem but I would prefer we hedge our bets ;)